src/config/server.rs

```diff
@@ -71,9 +73,21 @@ pub struct Server {
     /// Should the server serve the frontend assets in the `dist` directory?
     pub serve_dist: bool,
 
-    /// Should the server serve the frontend `index.html` for all
-    /// non-API requests?
-    pub serve_html: bool,
+    /// If set the server serves the frontend `index.html` for all
+    /// non-API requests using the Jinja template at the given path.
+    /// Setting this parameter requires setting
+    /// [`Self::og_image_base_url`] as well.
+    pub index_html_template_path: Option<PathBuf>,
```

Comment:
do we need this to be configurable?

Reply:
I'm not sure, actually. It depends on the exact file structure. It that's stable, we can just hard code `dist/index.html`

Reply:
thanks! I think since it was already hardcoded before that makes the PR a little more digestable. we can always change it later if there is a need for it :)